luxonis_ml/data/datasets/annotation.py

```diff
@@ -505,114 +442,83 @@ class ArrayAnnotation(Annotation):
-
     @staticmethod
+    @override
     def combine_to_numpy(
-        annotations: List["ArrayAnnotation"],
-        class_mapping: Dict[str, int],
-        **_,
+        annotations: List["ArrayAnnotation"], classes: List[int], _: int
     ) -> np.ndarray:
         out_arr = np.zeros(
             (
                 len(annotations),
-                len(class_mapping),
+                len(classes),
                 *np.load(annotations[0].path).shape,
             )
         )
         for i, ann in enumerate(annotations):
-            class_ = class_mapping.get(ann.class_, 0)
-            out_arr[i, class_] = np.load(ann.path)
+            out_arr[i, classes[i]] = np.load(ann.path)
         return out_arr
 
     @field_serializer("path")
     def serialize_path(self, value: FilePath) -> str:
         return str(value)
 
 
-class LabelAnnotation(Annotation):
-    """A custom unspecified annotation with a single primitive value.
-
-    @type value: Union[bool, int, float, str]
-    @ivar value: The value of the annotation.
-    """
-
-    type_: Literal["label"] = Field("label", alias="type")
-
-    value: Union[bool, int, float, str]
-
-    _label_type = LabelType.LABEL
-
-    @staticmethod
-    def combine_to_numpy(
-        annotations: List["LabelAnnotation"],
-        class_mapping: Dict[str, int],
-        **_,
-    ) -> np.ndarray:
-        out_arr = np.zeros((len(annotations), len(class_mapping))).astype(
-            type(annotations[0].value)
-        )
-        for i, ann in enumerate(annotations):
-            class_ = class_mapping.get(ann.class_, 0)
-            out_arr[i, class_] = ann.value
-        return out_arr
+class DatasetRecord(BaseModelExtraForbid):
+    files: Dict[str, FilePath]
+    annotation: Optional[Detection] = None
+    task_name: str = Field(validation_alias=AliasChoices("task", "task_name"))
 
+    @property
+    def file(self) -> FilePath:
+        if len(self.files) != 1:
+            raise ValueError("DatasetRecord must have exactly one file")
+        return next(iter(self.files.values()))
 
-class DatasetRecord(BaseModelExtraForbid):
-    """A record of an image and its annotation.
+    @model_validator(mode="before")
+    @classmethod
+    def validate_files(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if "file" in values:
+            values["files"] = {"image": values.pop("file")}
+        return values
 
-    @type file: FilePath
-    @ivar file: A path to the image.
-    @type annotation: Optional[Annotation]
-    @ivar annotation: The annotation for the image.
-    """
+    @model_validator(mode="before")
+    @classmethod
+    def validate_task_name(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if "task_name" not in values and "task" not in values:
+            if "segmentation" in values.get("annotation", {}):
+                values["task_name"] = "segmentation"
+            else:
+                values["task_name"] = "detection"
+        return values
```

Comment:
Is this us trying to automaticaly set task_name if not present based on "guessing" depending on what other fields are present? Should we rather make this field required?
Also would now be a good time to pick either only "task" or "task_name" and not support both going forward?

Reply:
Yeah. We need to somehow keep in mind that segmentation is usually not instance based so it has to be its own task. 
We can make the task required. It just wasn't before so I didn't want to change it. Also usually you're only using datasets with a single task so it's nice that you don't have to think of tasks in that case.

Reply:
Hm yeah agree that it is nice when it autopopulates since most of the time there is only one task. I fine with it keeping it not required if we are able to infer this value reliably. Or if it is ambiguous then warn/error out (probably want to error in this case)